### DISCUSSION TITLE
Is it possible to keep pressing a MIDI button to fast scroll menu items ?

### DISCUSSION THREAD

**Discussion by Onosendaifr** [probonopd/MiniDexed#678] [Q&A]
Navigating into menu could be quite cumbersome as soon as you have plenty of items.

For exemple, setting reverb size, there is 99 steps, You need to press 99 times the MIDI button to go back to 0 :)

I also have a library of 1300 voices, choosing one of them is a nightmare without autoscrolling, could it be added ?

**Comment by probonopd** [comment]
So you are using [MIDI Button Navigation](https://github.com/probonopd/MiniDexed/wiki/Files#midi-button-navigation), correct?

MIDI Button Navigation was implemented by @diyelectromusic in https://github.com/probonopd/MiniDexed/pull/365. What do you think about this idea Kevin? Would it be doabe? (While we are at it, why limit it to MIDI buttons; could we do the same for normal buttons (connected to GPIO pins) as well?

**Reply by Onosendaifr** [reply]
Correct !

**Comment by Onosendaifr** [comment]
I hope you will be able to add it, because without a rotary encoder is quite impossible to edit fast anything. 

When the "wrap" will be enable everywhere, it will better because we will be able to switch from 0 to 99 with only one push to the left. Still, a "fast travel" is really needed for a FM synth IMHO

**Comment by diyelectromusic** [comment]
It's all probably possible, but I guess the key thing is how many users do significant editing this way anyway vs using something like Dexed on a PC.

We've surfaced all the menus, but I'd not suggest they're a particularly nice way to set things up :)

Kevin

**Reply by Onosendaifr** [reply]
> the key thing is how many users do significant editing this way anyway vs using something like Dexed on a PC.

It's biased :) If people are choosing a PC to do it, maybe they do because the actual firmware is too cumbersome to use. There is space to improve UI/UX, and fast scroll is in the basic core of any synth.

**Reply by diyelectromusic** [reply]
<shrugs>

There is always room for improvement, but there is already a fully graphically edited microcontroller version - see MicroDexed touch - or the full UI/UX of a PC via MIDI.

But it all takes time and development, and we don't need to make all the projects the same.

I'm still in favour of exposing the main control interface over SysEx then people can experiment what whatever UI/UX they want to develop themselves.

But I agree it would be useful for a long-press to auto increment/decrement.  Once the current outstanding changes are done I'll see what is involved in adding a new action or event somehow.

Kevin
